ViAn/Filehandler/project.cpp

```diff
@@ -19,18 +20,40 @@ Project::Project(){
     this->files = new ProjFiles();
     this->m_name = "";
     this->m_id = -1;
-    this->m_videos.clear();
+    this->v_id = 0;
+    this->videos.clear();
+}
+/**
+ * @brief Project::~Project
+ * Clears contents of video map
+ */
+Project::~Project(){
+    for (auto vidIt = this->videos.begin(); vidIt != this->videos.end(); ++vidIt) {
+        delete vidIt->second;
+    }
 }
 
 /**
-* @brief Project::add_video
-* @param vid
-* add given video to project
-*/
-void Project::add_video(Video* vid)
-{
-    this->m_videos.push_back(vid);
+ * @brief Project::remove_video
+ * @param id
+ */
+void Project::remove_video(ID id){
```

Comment:
Here is another one that has the same function name in a different class. This is a reason why you need commenting.